.github/workflows/version.yml

```diff
@@ -1,123 +1,115 @@
           echo "NEXT_VERSION=$NEXT_VERSION" >> $GITHUB_ENV
-          echo "NEXT_VERSION_UNDERSCORE=$NEXT_VERSION_UNDERSCORE" >> $GITHUB_ENV
-          echo "NEXT_VERSION_ID=$NEXT_VERSION_ID" >> $GITHUB_ENV
-      - uses: actions/checkout@v2
+
+      - id: create-issue
+        uses: actions/github-script@v6.4.0
+        with:
+          script: |
+            const { data: issue }= await github.rest.issues.create({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              title: "Increment version for ${{ env.NEXT_VERSION }}",
+              body: "Issue for tracking the changes associated with a version increment."
+            });
+            console.error(JSON.stringify(issue));
+            return issue.number;
+          result-encoding: string
+            
+      - uses: actions/checkout@v3
         with:
           ref: ${{ env.BASE }}
-          token: ${{ steps.github_app_token.outputs.token }}
 
-      - name: Increment Patch Version
-        run: |
-          echo Incrementing $CURRENT_VERSION to $NEXT_VERSION
-          echo "  - \"$CURRENT_VERSION\"" >> .ci/bwcVersions
-          sed -i "s/opensearch        = $CURRENT_VERSION/opensearch        = $NEXT_VERSION/g" buildSrc/version.properties
-          echo Adding $NEXT_VERSION_UNDERSCORE after $CURRENT_VERSION_UNDERSCORE
-          sed -i "s/public static final Version $CURRENT_VERSION_UNDERSCORE = new Version(\([[:digit:]]\+\)\(.*\));/\0\n    public static final Version $NEXT_VERSION_UNDERSCORE = new Version($NEXT_VERSION_ID\2);/g" server/src/main/java/org/opensearch/Version.java
-          sed -i "s/CURRENT = $CURRENT_VERSION_UNDERSCORE;/CURRENT = $NEXT_VERSION_UNDERSCORE;/g" server/src/main/java/org/opensearch/Version.java
-
-      - name: Create Pull Request
-        uses: peter-evans/create-pull-request@v3
+      - name: Increment Patch Version on Major.Minor branch
+        uses: peternied/opensearch-core-version-updater@main
         with:
-          token: ${{ steps.github_app_token.outputs.token }}
-          base: ${{ env.BASE }}
-          branch: 'create-pull-request/patch-${{ env.BASE }}'
-          commit-message: Increment version to ${{ env.NEXT_VERSION }}
-          signoff: true
-          delete-branch: true
-          labels: |
-            autocut
-          title: '[AUTO] Increment version to ${{ env.NEXT_VERSION }}.'
-          body: |
-            I've noticed that a new tag ${{ env.TAG }} was pushed, and incremented the version from ${{ env.CURRENT_VERSION }} to ${{ env.NEXT_VERSION }}.
+          previous-version: ${{ env.CURRENT_VERSION }}
+          new-version: ${{ env.NEXT_VERSION }}
+          update-current: true
+
+      - run: git diff
+      - run: |
+          git config --global user.name 'Peter Nied'
+          git config --global user.email 'petern@amazon.com'
+          git add .
+          git commit -s -m "Increment version to $NEXT_VERSION #${{steps.create-issue.outputs.result}}"
```

Comment:
I'm curious to know how this commit will be signed-off. Do you set the GPG or RSA signing key somewhere? OR does it sign-off in a completely different way?

Reply:
`-s` handles the sign off, it looks like the following, see an example commit https://github.com/peternied/OpenSearch-1/issues/66
```
Add patch version 2.7.3
Signed-off-by: Peter Nied <petern@amazon.com>
```

Reply:
nice!